geotiff: CI failing on main — max_cloud_bytes missing from _CANONICAL_ORDER

## Summary

CI on `main` fails on every Python version because `test_open_geotiff_defines_canonical_order` rejects the current `open_geotiff` signature. The new `max_cloud_bytes` kwarg (added by #1932 / commit `8e61c14`) lives between `max_pixels` and `on_gpu_failure` in the function signature, but the test's `_CANONICAL_ORDER` constant in `xrspatial/geotiff/tests/test_reader_kwarg_order_1935.py:33` was not updated to match.

## Failing assertion (from CI run 25926000277, ubuntu-3.14)

```
AssertionError: open_geotiff kw-only params
  ['dtype', 'window', 'overview_level', 'band', 'name', 'chunks', 'gpu',
   'max_pixels', 'max_cloud_bytes', 'on_gpu_failure', 'missing_sources']
no longer match the canonical order
  ['dtype', 'window', 'overview_level', 'band', 'name', 'chunks', 'gpu',
   'max_pixels', 'on_gpu_failure', 'missing_sources'].
Update both the function and the _CANONICAL_ORDER constant together.
```

The test is doing its job: it fires when the signature drifts from the canonical list. The fix is to add `max_cloud_bytes` to the constant.

This is also blocking every other open PR (#1947, #1953, etc.) because their CI runs hit the same pre-existing failure on rebased `main`.

## Fix

`xrspatial/geotiff/tests/test_reader_kwarg_order_1935.py:33-44`: insert `"max_cloud_bytes"` between `"max_pixels"` and `"on_gpu_failure"` in `_CANONICAL_ORDER`.

--- a/xrspatial/geotiff/tests/test_reader_kwarg_order_1935.py
+++ b/xrspatial/geotiff/tests/test_reader_kwarg_order_1935.py
@@ -39,6 +39,7 @@ _CANONICAL_ORDER = (
     "chunks",
     "gpu",
     "max_pixels",
+    "max_cloud_bytes",
     "on_gpu_failure",
     "missing_sources",
 )

